### DISCUSSION TITLE
Displaying a QRCode as a Static widget

### DISCUSSION THREAD

**Discussion by labo-China** [Textualize/textual#2821] [Q&A] [answered]
I am making an application which allows user to login by scanning QRCode on the Screen.
In general, it generate a QRCode object using `qrcode` lib and convert the output to special characters and display them.

However, I found that the QRcode is not correctly displayed. This still happens even there`s only one Static object on the screen.

The original QRcode should be like this:
![image](https://github.com/Textualize/textual/assets/62234752/e7da39a7-9d83-4e11-9ecb-b26143b51c9f)

However, it displays like this using Textual:
![image](https://github.com/Textualize/textual/assets/62234752/91102cb7-4fdc-4735-81d2-79fb43799348)

The character output displays fine by simply using print() function.
What`s wrong with it? How should I get it right?

**Comment by willmcgugan** [comment, marked as answer]
I'm guessing you have escape sequences in your qrcode string. Try running it through `Text.from_ansi` before printing.

Please post code if you need further help.

**Reply by labo-China** [reply]
Your solution works perfect for me! Thanks for helping me!
